internal/pkg/cli/svc_deploy.go

```
@@ -489,17 +492,6 @@ func validateWorkloadManifestCompatibilityWithEnv(ws wsEnvironmentsLister, env v
 }
 
 func (o *deploySvcOpts) uriRecommendedActions() ([]string, error) {
-	type reachable interface {
```

Why do we remove this 👀 ? Will we insist on grabbing the URI for a worker service for example?

---

because `NewReachableService` will handle the logic and exit gracefully

---

```go
// NewReachableService returns a ReachableService based on the type of the service.
func NewReachableService(app, svc string, store ConfigStoreSvc) (ReachableService, error) {
	cfg, err := store.GetWorkload(app, svc)
	if err != nil {
		return nil, err
	}
	in := NewServiceConfig{
		App:         app,
		Svc:         svc,
		ConfigStore: store,
	}
	switch cfg.Type {
	case manifestinfo.LoadBalancedWebServiceType:
		return NewLBWebServiceDescriber(in)
	case manifestinfo.RequestDrivenWebServiceType:
		return NewRDWebServiceDescriber(in)
	case manifestinfo.BackendServiceType:
		return NewBackendServiceDescriber(in)
	default:
		return nil, fmt.Errorf("service %s is of type %s which cannot be reached over the network", svc, cfg.Type)
	}
}
```

```go
describer, err := describe.NewReachableService(o.appName, o.name, o.store)
if err != nil {
    return nil, err
} 
```

Looks like we will error out in this case! Should we handle detect ` fmt.Errorf("service %s is of type %s which cannot be reached over the network", svc, cfg.Type)` error here and swallow it?

---

Oh that's right we should swallow the error